src/main/java/pl/drivewheelsdeals/app/controller/ProductController.java

```diff
@@ -31,7 +41,116 @@ public Page<Product> listProducts(Integer page, Integer size) {
         return productService.getAllProducts(pageRequest);
     }
 
+    @PostMapping("/product/add")
+    public ProductOperationResponse addProduct(@Valid @RequestBody ProductOperationRequest request) throws BadRequestException {
```

Comment:
You used `@valid` annotation, but you did not provide any validation in model. Ex. NotBlank, minLenght etc.